contracts/managers/allocation-pricers/BinaryAllocationPricer.sol

```diff
@@ -0,0 +1,461 @@
+/*
+    Copyright 2019 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.5.7;
+pragma experimental "ABIEncoderV2";
+
+import { ERC20Detailed } from "openzeppelin-solidity/contracts/token/ERC20/ERC20Detailed.sol";
+import { Math } from "openzeppelin-solidity/contracts/math/Math.sol";
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import { ICore } from "set-protocol-contracts/contracts/core/interfaces/ICore.sol";
+import { ISetToken } from "set-protocol-contracts/contracts/core/interfaces/ISetToken.sol";
+import { SetTokenLibrary } from "set-protocol-contracts/contracts/core/lib/SetTokenLibrary.sol";
+
+import { FlexibleTimingManagerLibrary } from "../lib/FlexibleTimingManagerLibrary.sol";
+import { IAllocationPricer } from "./IAllocationPricer.sol";
+import { IOracle } from "../../meta-oracles/interfaces/IOracle.sol";
+
+
+/**
+ * @title BinaryAllocationPricer
+ * @author Set Protocol
+ *
+ * Implementing IAllocationPricer the BinaryAllocationPricer flips between two all or nothing
+ * allocations of the base asset depending what allocation the calling manager is seeking. In
+ * addition, if either collateral Set becomes 4x more valuable than the other the contract will
+ * create a new collateral Set and use that Set going forward.
+ */
+contract BinaryAllocationPricer is
+    IAllocationPricer
+{
+    using SafeMath for uint256;
+
+    /* ============ Structs ============ */
+    struct CollateralSetInfo {
+        ISetToken setInstance;
+        uint256 componentPrice;
+        uint256 componentDecimals;
+        uint256 collateralValue;
+    }
+
+    /* ============ Constants ============ */
+    uint256 constant SET_TOKEN_DECIMALS = 10 ** 18;
+    uint256 constant MINIMUM_COLLATERAL_NATURAL_UNIT = 10 ** 6;
+    uint256 constant ALLOCATION_PRICE_RATIO_LIMIT = 4;
+
+    /* ============ State Variables ============ */
+    ICore public coreInstance;
```

Comment:
At some point, i feel like we're better off dropping the `Instance`

Reply:
Feel like we need to choose a naming convention and stick with it. We did it because we thought it was a meaningful differentiation from an `address`...has that changed?